src/target/stm32mp15.c

```diff
@@ -157,13 +260,20 @@ static bool stm32mp15_attach(target_s *const target)
 	/* Disable C-Sleep, C-Stop, C-Standby for debugging */
 	target_mem_write32(target, DBGMCU_CTRL, DBGMCU_CTRL_DBGSLEEP | DBGMCU_CTRL_DBGSTOP | DBGMCU_CTRL_DBGSTBY);
 
+	/* Reference the APB-D in target storage for 0xe0000000 region manipulations */
+	stm32mp15_setup_apbd_ap(target);
 	return true;
 }
 
 static void stm32mp15_detach(target_s *const target)
```

Comment:
Given the Cortex-A attach/detach functions introduced above, we suggest these get refactored in name to include that they're specifically for the Cortex-M core - to help reduce code soupiness and confusion.

Reply:
They're not CM4-specific, SoC SWO the peripheral is shared; but I don't see it being useful for CA7 (yet) and I wrote this PR such that it reuses target_storage for AXI-AP, so those are exclusive. Renaming.

Reply:
The statement that it's core specific comes from it calling `cortexm_` functions directly. It would require extra work to make it generic to both cores (probably in the form of having the core common code in one function and having a `stm32mp15_cm4_detach()` and a `stm32mp15_ca7_detach()` that call that common routine and the core-type-specific detach routines. Thank you for doing the rename.